### PR TITLE
remove max_matrix_locations and introduce max_matrix_location_pairs

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes https://github.com/valhalla/valhalla/issues/3566

named it `max_matrix_locations_total`, ok with you? should I add a small test in `matrix.cc`?

EDIT: dumb name, instead @kevinkreiser original suggestion `max_matrix_location_pairs`

### CODE REVIEW

**Comment by kevinkreiser** [comment]
i mean im no word smith here but `max_matrix_locations_total` isn't the total number of locations, its the number of location pairs if i understood your issue correctly. am i misunderstanding here?

File: scripts/valhalla_build_config [hunk 1]
```diff
@@ -445,51 +445,51 @@ help_text = {
       'max_distance': 'Maximum b-line distance between all locations in meters',
       'max_locations': 'Maximum number of input locations',
       'max_matrix_distance': 'Maximum b-line distance between 2 most distant locations in meters for a matrix',
-      'max_matrix_locations': 'Maximum number of sources or targets for a matrix'
+      'max_matrix_locations_total': 'Maximum number of routes computed with the matrix, e.g. 2500 = 50:50 or 1:2500'
```
Comment (kevinkreiser) [inline scripts/valhalla_build_config:448]: :+1: on the description

**Comment by nilsnolde** [comment]
😅point taken. Pairs it is

**Review by kevinkreiser — APPROVED** [review]

**Comment by nilsnolde** [comment]
uch, can you approve again pls?

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**